integration-tests/tests/src/tests/queries.ts

```diff
@@ -15,11 +15,841 @@
+    expect(decimal128Objects.filtered("decimal128Col != 10002").length).equals(2);
+
+    realm.close();
+  });
+
+  it("should support queries with UUID as primary key", () => {
+    const testStringUuids = [
+      "01b1a58a-bb92-47a2-a3aa-d9c735e6fd42",
+      "ab01fec2-55d5-4fac-9e04-980bff6a521d",
+      "6683f348-d441-4846-81aa-cc375b771032",
+    ];
+    const nonExistingStringUuid = "aaaaaaaa-aaaa-4aaa-aaaa-aaaaaaaaaaaa";
+
+    const primUuidSchema = {
+      name: "PrimUuidObject",
+      primaryKey: "_id",
+      properties: {
+        _id: "uuid",
+        text: "string",
+      },
+    };
+
+    const realm = new Realm({ schema: [primUuidSchema] });
+    realm.write(() => {
+      testStringUuids.forEach((uuidStr) => {
+        realm.create(primUuidSchema.name, { _id: new UUID(uuidStr), text: uuidStr });
+      });
+    });
+
+    // objectForPrimaryKey tests
+    const nonExisting = realm.objectForPrimaryKey(primUuidSchema.name, new UUID(nonExistingStringUuid));
+    expect(nonExisting).equals(
+      null,
+      `objectForPrimaryKey should return undefined for new UUID("${nonExistingStringUuid}")`,
+    );
+
+    testStringUuids.forEach((uuidStr) => {
+      const obj = realm.objectForPrimaryKey(primUuidSchema.name, new UUID(uuidStr));
+      expect(obj).not.equal(null, `objectForPrimaryKey should return a Realm.Object for new UUID("${uuidStr}")`);
+      //@ts-expect-error _id is part of schema.
+      expect(obj._id.toString()).equals(uuidStr);
+    });
+
+    // results.filtered tests
+    const emptyFiltered = realm.objects(primUuidSchema.name).filtered("_id == $0", new UUID(nonExistingStringUuid));
+    expect(emptyFiltered.length).equals(
+      0,
+      `filtered objects should contain 0 elements when filtered by new UUID("${nonExistingStringUuid}")`,
+    );
+
+    testStringUuids.forEach((uuidStr) => {
+      const filtered = realm.objects(primUuidSchema.name).filtered("_id == $0", new UUID(uuidStr));
+      expect(filtered.length).equals(1, `filtered objects should contain exactly 1 of new UUID("${uuidStr}")`);
+    });
+  });
+
+  it("should work with between", () => {
+    const realm = new Realm({ schema: [IntObjectSchema] });
+    realm.write(() => {
+      for (let i = 0; i < 10; i++) {
+        realm.create(IntObjectSchema.name, { intCol: i });
+      }
+    });
+
+    const range = realm.objects<IntObject>(IntObjectSchema.name).filtered("intCol BETWEEN {5, 8}"); // 5, 6, 7, 8
+    expect(range.length).equals(4);
+    expect(range[0].intCol).equals(5);
+    expect(range[1].intCol).equals(6);
+    expect(range[2].intCol).equals(7);
+    expect(range[3].intCol).equals(8);
+
+    realm.close();
+  });
+});
+
+import { IPerson, PersonSchema } from "../schemas/person-and-dogs";
+import { IContact, ContactSchema } from "../schemas/contact";
 
-import { IPerson, Person, PersonSchema } from "../schemas/person-and-dogs";
-import { IContact, Contact, ContactSchema } from "../schemas/contact";
 interface IPrimitive {
```

Comment:
I left the old tests basically untouched and separate but I could also incorporate them in a larger describe.

Reply:
I think it's fine to leave them for now.  Maybe just add a comment about how these tests will be combined with the above in a later refactoring effort.